documentation/sphinx/source/developer-guide.rst

```diff
@@ -756,6 +756,118 @@ If you only need to detect the *fact* of a change, and your response doesn't dep
+
+Each special key that existed before api version 630 is its own module. These are
+
+#. ``\xff\xff/cluster_file_path`` Path to this database's cluster file
+#. ``\xff\xff/connection_string`` The desired contents of the cluster file. Should only be different if the client does not have permission to write to the cluster file.
+#. ``\xff\xff/status/json`` See :doc:`Machine-readable status <mr-status>`
+
+Prior to api version 630, it was also possible to read a range starting at
+``\xff\xff/worker_interfaces``. This is mostly an implementation detail of fdbcli,
+but it's available in api version 630 as a module with prefix ``\xff\xff/worker_interfaces/``.
+
+Api version 630 includes two new modules with prefixes
+``\xff\xff/transaction/`` (information about the current transaction), and
+``\xff\xff/metrics/`` (various metrics, not transactional).
+
+Transaction module
+------------------
+
+Reads from the transaction module generally do not require an rpc and only inspect in-memory state for the current transaction.
+
+There are three sets of keys exposed by the transaction module, and each set uses the same encoding, so let's first describe that encoding.
+
+Let's say we have a set of keys represented as intervals of the form ``begin1 <= k < end1 && begin2 <= k < end2 && ...``.
+It could be the case that some of the intervals overlap, e.g. if ``begin1 <= begin2 < end1``, or are adjacent, e.g. if ``end1 == begin2``.
+If we merge all overlapping/adjacent intervals then sort, we end up with a canonical representation of this set of keys.
+
+We encode this canonical set as ordered key value pairs like this::
+
+  <namespace><begin1> -> "1"
+  <namespace><end1> -> "0"
+  <namespace><begin2> -> "1"
+  <namespace><end2> -> "0"
+  ...
+
+Python example::
+
+  >>> tr = db.create_transaction()
+  >>> tr.add_read_conflict_key('foo')
+  >>> tr.add_read_conflict_range('bar/', 'bar0')
+  >>> for k, v in tr.get_range_startswith('\xff\xff/transaction/read_conflict_range/'):
+  ...     print(k, v)
+  ...
+  ('\xff\xff/transaction/read_conflict_range/bar/', '1')
+  ('\xff\xff/transaction/read_conflict_range/bar0', '0')
+  ('\xff\xff/transaction/read_conflict_range/foo', '1')
+  ('\xff\xff/transaction/read_conflict_range/foo\x00', '0')
+
+For read-your-writes transactions, this canonical encoding of conflict ranges
+is already available in memory, and so requesting small ranges is
+correspondingly cheaper than large ranges.
+
+For transactions with read-your-writes disabled, this canonical encoding is computed on
+every read, so you're paying the full cost in CPU time whether or not you
+request a small range.
+
+The namespaces for sets of keys are
+
+#. ``\xff\xff/transaction/read_conflict_range/`` This is the set of keys that will be used for read conflict detection. If another transaction writes to any of these keys after this transaction's read version, then this transaction won't commit.
```

Comment:
Looks like you are explaining how these `read_conflict_range`s are used to detect conflicts. 
Maybe adding a link to the session `Conflict Detection` can save the words? 
Anyway, this short explanation is good enough.